test/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit.test.cpp

```diff
@@ -570,6 +576,195 @@ TEST(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit, ComputePasses)
     }
 }
 
+TEST(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit, ComputePassesWithModel)
+{
+    // undefined model
+    {
+        const Kepler kepler = {
+            COE::Undefined(),
+            Instant::Undefined(),
+            Earth::Default(),
+            Kepler::PerturbationType::None,
+        };
+        const Interval interval = Interval::Closed(Instant::Now(), Instant::Now() + Duration::Days(1.0));
+        EXPECT_THROW(Orbit::ComputePassesWithModel(kepler, interval), ostk::core::error::runtime::Undefined);
+    }
+
+    // undefined interval
+    {
+        const TLE tle =
+            TLE("1 43890U 18111Q   20195.55622117 +.00000241 +00000-0 +28512-4 0  9991",
+                "2 43890 097.6899 099.9703 0003551 181.1072 179.0140 14.92932318084236");
+        const SGP4 sgp4 = SGP4(tle);
+        EXPECT_THROW(Orbit::ComputePassesWithModel(sgp4, Interval::Undefined()), ostk::core::error::runtime::Undefined);
+    }
+
+    // Kepler model
+    {
+        // Orbit setup
+
+        const Length semiMajorAxis = Length::Kilometers(7000.0);
+        const Real eccentricity = 0.0;
+        const Angle inclination = Angle::Degrees(45.0);
+        const Angle raan = Angle::Degrees(0.0);
+        const Angle aop = Angle::Degrees(0.0);
+        const Angle trueAnomaly = Angle::Degrees(0.0);
+
+        const COE coe = {semiMajorAxis, eccentricity, inclination, raan, aop, trueAnomaly};
+
+        const Instant epoch = Instant::DateTime(DateTime(2018, 1, 1, 0, 0, 0), Scale::UTC);
+        const Derived gravitationalParameter = EarthGravitationalModel::EGM2008.gravitationalParameter_;
+        const Length equatorialRadius = EarthGravitationalModel::EGM2008.equatorialRadius_;
+        const Real J2 = EarthGravitationalModel::EGM2008.J2_;
+        const Real J4 = EarthGravitationalModel::EGM2008.J4_;
+
+        const Kepler keplerianModel = {
+            coe, epoch, gravitationalParameter, equatorialRadius, J2, J4, Kepler::PerturbationType::None
+        };
+
+        // partial
+        {
+            // Pass test
+            const Instant startInstant = Instant::DateTime(DateTime::Parse("2018-01-01 00:00:00"), Scale::UTC);
+            const Instant endInstant = Instant::DateTime(DateTime::Parse("2018-01-01 02:30:00"), Scale::UTC);
+            const Interval interval = Interval::Closed(startInstant, endInstant);
+
+            const Array<Pass> passes = Orbit::ComputePassesWithModel(keplerianModel, interval);
+
+            for (const auto &pass : passes)
+            {
+                EXPECT_TRUE(pass.isDefined());
+            }
+            EXPECT_EQ(passes.accessLast().getType(), Pass::Type::Partial);
```

Comment:
I thought you wanted to drop support for partial passes, will this be in a followup MR?